core/types/transaction_signing.go

```diff
@@ -126,27 +129,28 @@ func (s EIP2718Signer) Sender(tx *Transaction) (common.Address, error) {
 		// ACL txs are defined to use 0 and 1 as their recovery id, add
 		// 27 to become equivalent to unprotected Homestead signatures.
 		V = new(big.Int).Add(V, big.NewInt(27))
+	default:
+		return common.Address{}, ErrTxTypeNotSupported
 	}
 	if tx.ChainId().Cmp(s.chainId) != 0 {
 		return common.Address{}, ErrInvalidChainId
 	}
 	return recoverPlain(s.Hash(tx), R, S, V, true)
 }
 
-// SignatureValues returns signature values.
 func (s EIP2718Signer) SignatureValues(tx *Transaction, sig []byte) (R, S, V *big.Int, err error) {
-	R, S, V, err = HomesteadSigner{}.SignatureValues(tx, sig)
-	if err != nil {
-		return nil, nil, nil, err
-	}
 	switch tx.Type() {
 	case LegacyTxType:
+		R, S, V = decodeSignature(sig)
```

Comment:
Wouldn't it be less copy-pastey if this just did `return s.EIP155Signer.SignatureValues(tx, sig)` ?